### PR TITLE
Refactor/env var consolidation 4

### PR DESCRIPTION
**Pull request by hieuntg81**
## Summary

- Problem: 6 separate env vars for IAM/OIDC/Cogit/LLM URLs are redundant ? OIDC endpoints (`authorize`, `token`, `issuer`) are always derived from the IAM base, and API version paths (`/api/v1`) are baked into env vars instead of being appended at point of use.
- Why it matters: Adding a new IAM endpoint requires updating multiple env vars. Misconfiguring one OIDC URL while the others are correct causes subtle auth failures that are hard to diagnose.
- What changed:
  - Replaced 6 env vars with 3 base URL vars:
    - `BRV_API_BASE_URL`, `BRV_AUTHORIZATION_URL`, `BRV_ISSUER_URL`, `BRV_TOKEN_URL` ? `BRV_IAM_BASE_URL`
    - `BRV_COGIT_API_BASE_URL` ? `BRV_COGIT_BASE_URL`
    - `BRV_LLM_API_BASE_URL` ? `BRV_LLM_BASE_URL`
  - OIDC URLs (`authorizationUrl`, `issuerUrl`, `tokenUrl`) are now derived from `iamBaseUrl` inside `getCurrentConfig()`
  - API version paths (`/api/v1`) are appended at point of use in `feature-handlers.ts`
  - Updated `EnvironmentConfig` type, all consumers, and tests
- What did NOT change (scope boundary): No behavioral changes to auth flow, OIDC discovery, or any HTTP service. Only the source of URL values changed.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [x] Refactor (no behavior change)
- [ ] Documentation
- [ ] Test
- [ ] Chore (build, dependencies, CI)

## Scope (select all touched areas)

- [ ] TUI / REPL
- [ ] Agent / Tools
- [ ] LLM Providers
- [x] Server / Daemon
- [x] Shared (constants, types, transport events)
- [ ] CLI Commands (oclif)
- [ ] Hub / Connectors
- [ ] Cloud Sync
- [ ] CI/CD / Infra

## Linked issues

- Related #ENG-2002

## Root cause (bug fixes only, otherwise write `N/A`)

N/A

## Test plan

- Coverage added:
  - [x] Unit test
  - [ ] Integration test
  - [ ] Manual verification only
- Test file(s): `test/unit/config/environment.test.ts`
- Key scenario(s) covered:
  - All URL properties read correctly from new env vars
  - OIDC URLs derived correctly from `BRV_IAM_BASE_URL`
  - Missing required env var throws with correct variable name

## User-visible changes

- `.env.example` updated ? developers must update their `.env.development` / `.env.production` files:

| Old | New |
|---|---|
| `BRV_API_BASE_URL=http://localhost:3000/api/v1` | `BRV_IAM_BASE_URL=http://localhost:3000` |
| `BRV_AUTHORIZATION_URL=http://localhost:3000/api/v1/oidc/authorize` | *(removed, derived)* |
| `BRV_ISSUER_URL=http://localhost:3000/api/v1/oidc` | *(removed, derived)* |
| `BRV_TOKEN_URL=http://localhost:3000/api/v1/oidc/token` | *(removed, derived)* |
| `BRV_COGIT_API_BASE_URL=http://localhost:3001/api/v1` | `BRV_COGIT_BASE_URL=http://localhost:3001` |
| `BRV_LLM_API_BASE_URL=http://localhost:3002` | `BRV_LLM_BASE_URL=http://localhost:3002` |

## Evidence

- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording

TypeScript type check and all environment config tests pass after changes.

## Checklist

- [x] Tests added or updated and passing (`npm test`)
- [x] Lint passes (`npm run lint`)
- [x] Type check passes (`npm run typecheck`)
- [ ] Build succeeds (`npm run build`)
- [x] Commits follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [ ] Documentation updated (if applicable)
- [x] No breaking changes (or clearly documented above)
- [x] Branch is up to date with `main`

## Risks and mitigations

- Risk: Developers with existing `.env.development` / `.env.production` files will get startup errors after pulling this change.
  - Mitigation: Error message clearly states which env var is missing (`Missing required environment variable: BRV_IAM_BASE_URL`). `.env.example` serves as reference.

### CODE REVIEW

File: src/server/config/environment.ts [hunk 1]
```diff
@@ -42,28 +49,40 @@ const DEFAULTS = {
...
-  hubRegistryUrl: DEFAULTS.hubRegistryUrl,
-  issuerUrl: readRequiredEnv('BRV_ISSUER_URL'),
-  llmApiBaseUrl: readRequiredEnv('BRV_LLM_API_BASE_URL'),
-  scopes: [...DEFAULTS.scopes[ENVIRONMENT]],
-  tokenUrl: readRequiredEnv('BRV_TOKEN_URL'),
-  webAppUrl: readRequiredEnv('BRV_WEB_APP_URL'),
-})
+export const getCurrentConfig = (): EnvironmentConfig => {
+  const iamBaseUrl = readRequiredEnv('BRV_IAM_BASE_URL')
+  const oidcBase = `${iamBaseUrl}/api/v1/oidc`
```
Comment (leehpham) [inline src/server/config/environment.ts]: ### `/api/v1` is hardcoded in two places with split semantics

The stated goal of this refactor is to move `/api/v1` to "point of use", but it is baked into config at construction time here (`/api/v1/oidc`) while also being appended at service instantiation time in `feature-handlers.ts` (`const API_V1 = '/api/v1'`).

If the API version changes, you'd need to find and update both locations — and they follow different patterns (one in config derivation, one at service wiring). The comment says OIDC is "a fixed, auth-specific structure" but `/api/v1` is a versioning prefix, not auth-specific.

Consider extracting the version prefix as a shared constant so both locations reference the same source of truth.

File: src/server/config/environment.ts [hunk 2]
```diff
@@ -42,28 +49,40 @@ const DEFAULTS = {
   },
 } as const
 
+const normalizeUrl = (url: string): string => (url.endsWith('/') ? url.slice(0, -1) : url)
```
Comment (leehpham) [inline src/server/config/environment.ts]: ### `normalizeUrl` only strips a single trailing slash

```ts
const normalizeUrl = (url: string): string => (url.endsWith('/') ? url.slice(0, -1) : url)
```

`https://iam.test//` → `https://iam.test/` (still has a trailing slash after normalization).

If you're going to normalize, do it properly:
```ts
const normalizeUrl = (url: string): string => url.replace(/\/+$/, '')
```

This handles any number of trailing slashes in a single pass.

File: src/server/infra/process/feature-handlers.ts [hunk 3]
```diff
@@ -102,9 +102,14 @@ export async function setupFeatureHandlers({
   const envConfig = getCurrentConfig()
   const tokenStore = createTokenStore()
   const projectConfigStore = new ProjectConfigStore()
-  const userService = new HttpUserService({apiBaseUrl: envConfig.apiBaseUrl})
-  const teamService = new HttpTeamService({apiBaseUrl: envConfig.apiBaseUrl})
-  const spaceService = new HttpSpaceService({apiBaseUrl: envConfig.apiBaseUrl})
+
+  // API version paths appended at point of use.
+  // Note: IAM and Cogit currently share this version path, but may version independently in the future.
```
Comment (leehpham) [inline src/server/infra/process/feature-handlers.ts:107]: ### `API_V1` is trapped in function scope

This constant is defined inside `setupFeatureHandlers()`. If any other module needs the same version prefix (and `environment.ts:70` already duplicates it as a string literal), it must redeclare the string.

Since the whole point of this refactor is to centralize URL construction, the version prefix should live in a shared location — the config module or a constants file — so all consumers reference a single source of truth.

File: test/unit/config/environment.test.ts [hunk 4]
```diff
@@ -123,13 +121,45 @@ describe('Environment Configuration', () => {
...
+      const config = getCurrentConfig()
 
-      expect(() => getCurrentConfig()).to.throw('Missing required environment variable: BRV_API_BASE_URL')
+      expect(config.iamBaseUrl).to.equal('https://iam.test')
+      expect(config.cogitBaseUrl).to.equal('https://cogit.test')
+      expect(config.llmBaseUrl).to.equal('https://llm.test')
+      expect(config.webAppUrl).to.equal('https://app.test')
+      expect(config.gitRemoteBaseUrl).to.equal('https://git.test')
+      expect(config.authorizationUrl).to.equal('https://iam.test/api/v1/oidc/authorize')
     })
```
Comment (leehpham) [inline test/unit/config/environment.test.ts:163]: ### `normalizeUrl` edge cases not covered

This test only covers the happy path (single trailing slash). Missing cases:
- **Multiple trailing slashes**: `https://iam.test//` — the current `normalizeUrl` implementation only strips one slash, so this would silently produce `https://iam.test/`
- **Base URL containing path segments**: `https://iam.test/api/v1/` — the most likely migration mistake (developer copies old `BRV_API_BASE_URL` value into `BRV_IAM_BASE_URL`), which would silently produce double-pathed OIDC URLs like `https://iam.test/api/v1/api/v1/oidc/authorize`
- **Empty string / whitespace-only values**

The migration-mistake scenario is particularly dangerous because it produces no error — just silent 404s at auth time.

**Review by leehpham — COMMENTED** [review]
Review comments on environment variable consolidation — design and test gaps.

**Review by leehpham — APPROVED** [review]
Looks good. Can consider deploying in the next release @bao-byterover

**Comment by bao-byterover** [comment]
@hieuntg81 LGTM.

Before merging, let’s re-check the plan to make sure this change is safe and won’t affect other PRs.

After it’s merged, we should also provide an updated .env for the team.

**Review by bao-byterover — APPROVED** [review]
LGTM @hieuntg81
